### ISSUE TITLE
injecting external-helpers in a node app

### ISSUE THREAD

**Issue by travi** [babel/babel#7357]
Choose one: is this a bug report or feature request? (docs?) bug report

I'm trying to use a package that assumes that the external-helpers are available as a global. From the [docs](https://babeljs.io/docs/plugins/external-helpers/#injecting-the-external-helpers), I should be able to inject them to `global` in my node app by using `require("babel-core").buildExternalHelpers();`. However, use of that still results in the following error: `ReferenceError: babelHelpers is not defined`

### Babel/Babylon Configuration (.babelrc, package.json, cli command)
Since the `buildExternalHelpers()` function needs to run before the package is imported and my app uses es module imports, I'm using a bootstrap file as an entry point that is ignored from transpilation and just tries to inject the helpers before loading the actual app:

```js
require("babel-core").buildExternalHelpers();
const app = require('./app');
```

### Expected Behavior

`babelHelpers` should be added to `global` so that it is available for the package that assumes it is available there.

from the docs:
> This injects the external helpers into `global`.

### Current Behavior

`babelHelpers` is not made available on `global`, resulting in `ReferenceError: babelHelpers is not defined`

### Possible Solution

The docs also [mention](https://babeljs.io/docs/plugins/external-helpers/#getting-the-external-helpers) generating a helpers file with `./node_modules/.bin/babel-external-helpers [options] > helpers.js`. It wasn't obvious to me that this file could be imported to accomplish the same goal as `buildExternalHelpers()` until I started reading the source of that file. Importing that file instead does work for my app. I'll need this file elsewhere, so I'll likely just continue importing that instead, even if there is a way to use `buildExternalHelpers()`.

With that approach, my bootstrap file has the following contents instead:

```js
require('../../vendor/babel-helpers');
const app = require('./app');
```

### Your Environment

| software         | version(s)
| ---------------- | -------
| Babel            | 6.26.0
| node             | 8.9.4
| npm              | 5.6.0
| Operating System | macOS High Sierra 

### Forum

While I was still trying to find a working solution, I was trying to find the best place to ask questions. The website still links to a discourse forum that no longer seems to exist. It'd be a good idea to either remove the link from the site or at least have it link to somewhere that describes the currently recommended approach for getting that type of help.

**Comment by Andarist** [comment]
`buildExternalHelpers` builds helpers as string, you should save the result in a file and later `require` that file in your app.

**Comment by travi** [comment]
@Andarist understood, which is what i ended up doing, but that is not clear from the docs:

> ```js
> require("babel-core").buildExternalHelpers();
> ```
>
> This injects the external helpers into global.

if what you mention is the intent, it seems like the docs should at least be updated to make it clear that the command above does not directly update `global`

**Comment by abiduzz420** [comment]
I shall take this up. This would be my first contribution to babel.

**Comment by debugpai2** [comment]
Is this claimed already? If not I would like to work on this

**Comment by abiduzz420** [comment]
Hi @debugpai2 I haven't been able to spend time working on this issue. Can I get the PR ready in a day or two ?

**Comment by debugpai2** [comment]
Hey @abiduzz420, any luck with the issue? Can I help with anything?

**Comment by debugpai2** [comment]
@Andarist What is expected here? for `require("babel-core").buildExternalHelpers();` not to return a string anymore or for the documentation to be updated?

**Comment by Andarist** [comment]
I think documentation must be updated + we could add new function, smth like `injectExternalHelpers` which could build the helpers and evaluate them in global context, although we should add discouraging note to it so people don't use this in production.

**Comment by debugpai2** [comment]
documentation is changed by someone else [in this PR](https://github.com/babel/website/pull/1584) already. I will work on adding the function `injectExternalHelpers`

**Comment by loganfsmyth** [comment]
I missed this when it originally got posted, sorry.

I'm not sure I support the idea of adding this as an `eval` in `babel-core`. If the issue is that it isn't clear how to use the existing function, let's absolutely fix that, but including this in `@babel/core` encourages people to have a runtime dependency on core, where normally it would be a devDependency, and if anyone happened to use that function as part of a client-side bundle, Webpack would end up trying to bundle up all of core into their application.

Let's make the documentation clearer, but I don't think `injectExternalHelpers` is an API we should recommend.

**Comment by travi** [comment]
> If the issue is that it isn't clear how to use the existing function, let's absolutely fix that

Yeah, that was my main intention of posting the issue. Sorry if it wasn't clear, but thats what I meant by suggesting it was a docs bug:

> Choose one: is this a bug report or feature request? (docs?) bug report

> including this in @babel/core encourages people to have a runtime dependency on core, where normally it would be a devDependency, and if anyone happened to use that function as part of a client-side bundle, Webpack would end up trying to bundle up all of core into their application.

that all makes sense and i think further highlights the importance of that docs update. the previous wording implied to me that it was encouraged to depend on it at runtime. generating my own vendored file that i manage isn't ideal either, so depending on a package seemed reasonable if supported. 

making the intent as clear as possible would save consumers from digging too deep in the wrong direction like i did.

**Comment by debugpai2** [comment]
Thanks @loganfsmyth for shedding some light on this. It makes sense now why this would be a bad practice. I would close my PR which was adding `babelHelpers` in the global scope. Meanwhile there is [this PR](https://github.com/babel/website/pull/1584) to update the docs. If someone can merge it, this issue can be closed.

**Comment by mk-pmb** [comment]
Looks like the "good first issue" tag here is outdated.

**Comment by nicolo-ribaudo** [comment]
Since `good first issue` is paired with `i: docs`, I think that the original intent was "Updating the docs would be a good first contribution"

**Comment by mk-pmb** [comment]
Indeed, however now it seems there's a PR that solves it. If not, someone should comment on #1584 what's missing.

**Comment by lidoravitan** [comment]
Can I help with it? @mk-pmb @nicolo-ribaudo

**Comment by mk-pmb** [comment]
@lidoravitan IMO we should wait until babel/website#1584 is resolved, then see what parts of this issue remain. Maybe you can find ways to help there.

**Comment by lidoravitan** [comment]
Sure @mk-pmb. it looks like I have to resolve the conflicts file, right? ([babel/website#1584](babel/website#1584 ))

**Comment by pree-T** [comment]
Is this issue still open?May I work on it? Can someone help me as I am new to open source contribution?

**Comment by deepuyadav004** [comment]
Hello everyone, I am new to open-source contribution and I am looking for some good first issues. Is this issue still open i would like to give my contribution.

**Comment by Arnavya** [comment]
I am new to open source, and this issue seems to be a good start. I would really appreciate the opportunity to contribute.

**Comment by mk-pmb** [comment]
@Arnavya Welcome to GitHub. All the permissions you'd need to contribute are granted by this project's license already, so feel free to start your attempt immediately.

**Comment by Mayur-Murarka** [comment]
Hello everyone, I am new to open-source contribution and I am looking for some good first issues. Is this issue still open i would like to give my contribution.

**Comment by LeonxLJX** [comment]
I'd like to take this one.

From the report, the useful first step is to pin down the exact code path involved before touching anything — these tend to have a small true cause behind a broad-looking symptom, and guessing at it usually produces a patch that masks the behavior instead of fixing it.

Plan: trace the path, reproduce it, then make a focused change with a test if this area has coverage. I'll keep the diff small so it's easy to review.

If there's an existing convention in this repo for how this kind of thing is handled, I'd rather follow it than invent my own — happy to be pointed at one.

(claiming via @LeonxLJX)
